clients/ui/frontend/src/app/api/modelCatalog/service.ts

```diff
@@ -87,3 +91,35 @@ export const getListCatalogModelArtifacts =
       }
       throw new Error('Invalid response format');
     });
+
+export const getPerformanceArtifacts =
+  (hostPath: string, queryParams: Record<string, unknown> = {}) =>
+  (
+    opts: APIOptions,
+    sourceId: string,
+    modelName: string,
+    params?: PerformanceArtifactsParams,
+    filterData?: ModelCatalogFilterStates,
+    filterOptions?: CatalogFilterOptionsList | null,
+  ): Promise<CatalogArtifactList> => {
+    const allParams: Record<string, unknown> = {
+      ...queryParams,
+      ...(params?.targetRPS !== undefined && { targetRPS: params.targetRPS }),
+      ...(params?.latencyProperty && { latencyProperty: params.latencyProperty }),
```

Comment:
we might need to add `rpsProperty, hardwareCountProperty, hardwareTypeProperty` here as we are passing it in params from hook